doc/cpp/hardware.rst

```diff
@@ -152,10 +152,10 @@ and handling exceptions on MPI exit.
         if the rank with the exception attempts to call :cpp:any:`MPI_Finalize` and
         other ranks are waiting in other MPI calls. If this happens inside a try-catch
         block, the deadlock stops the exception from being handled.
-        For this reason the destructor of :cpp:class:`with_mpi` only calls
+        For this reason, the destructor of :cpp:class:`with_mpi` only calls
         :cpp:any:`MPI_Finalize` if there are no uncaught exceptions.
-        This isn't perfect because the other MPI ranks still deadlock,
-        however it gives the exception handling code to report the error for debugging.
+        This isn't perfect because the other MPI ranks are still deadlock,
```

Comment:
Deadlock is used as a verb here ... not common outside of programmer jargon, but 'are' seems misplaced here. 'Can' could be an option.

Reply:
it is used many times in this file as a noun

Reply:
Yes, but here it is a verb. A deadlock is the state in which multiple threads wait for each other (often circular) and none can make progress. To deadlock is to diverge into that state. Lingo, as I said ;)